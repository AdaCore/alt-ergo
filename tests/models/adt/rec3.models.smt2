(set-option :produce-models true)
(set-logic ALL)
(declare-datatypes ((data1 0) (data2 0)) (
  ((cons1 (d1 data1)) (cons2 (d2 data2)))
  ((cons3 (d3 data1)) (cons4 (d4 Int) (d5 Int)))
))
(declare-const a data1)
(assert ((_ is cons1) a))
(check-sat)
(get-model)
