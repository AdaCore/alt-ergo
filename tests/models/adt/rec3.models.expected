
unknown
(
  (define-fun a () data1 (cons1 (cons2 (cons4 0 1))))
)
